src/bsoncxx/types/value.cpp

```diff
@@ -56,35 +74,52 @@ value& value::operator=(const value& rhs) {
 
 value::value(value&& rhs) noexcept {
     *this = std::move(rhs);
+    rhs._type = stdx::nullopt;
```

Comment:
What I meant wast that if value's move assignment operator is correctly implemented, than the result of executing it should be that rhs._type becomes nullopt already. You shouldn't need to do it again.